examples/pennylane_qubit_rotation.py

```diff
@@ -0,0 +1,367 @@
+                   -i \sin \frac{\phi_1}{2} &  \cos \frac{\phi_1}{2}
+    \end{bmatrix},
+
+and then around the y-axis via the gate
+
+.. math::
+    R_y(\phi_2) = e^{-i \phi_2 \sigma_y/2} =
+   \begin{bmatrix} \cos \frac{\phi_2}{2} &  - \sin \frac{\phi_2}{2} \\
+                   \sin \frac{\phi_2}{2} &  \cos \frac{\phi_2}{2}
+   \end{bmatrix}.
+
+After these operations the qubit is now in the state
+
+.. math::  | \psi \rangle = R_y(\phi_2) R_x(\phi_1) | 0 \rangle.
+
+Finally, we measure the expectation value :math:`\langle \psi \mid \sigma_z \mid \psi \rangle`
+of the Pauli-Z operator
+
+.. math::
+   \sigma_z =
+   \begin{bmatrix} 1 &  0 \\
+                   0 & -1
+   \end{bmatrix}.
+
+Using the above to calculate the exact expectation value, we find that
+
+.. math::
+    \braketT{\psi}{\sigma_z}{\psi}
+    = \braketT{0}{R_x(\phi_1)^\dagger R_y(\phi_2)^\dagger \sigma_z  R_y(\phi_2) R_x(\phi_1)}{0}
+    = \cos(\phi_1)\cos(\phi_2).
+
+Depending on the circuit parameters :math:`\phi_1` and :math:`\phi_2`, the
+output expectation lies between :math:`1` (if :math:`\ket{\psi} = \ket{0}`)
+and :math:`-1` (if :math:`\ket{\psi} = \ket{1}`).
+
+Let's see how we can easily implement and optimize this circuit using PennyLane.
+"""
+
+##############################################################################
+# Importing PennyLane and NumPy
+# -----------------------------
+#
+# The first thing we need to do is import PennyLane, as well as the wrapped version
+# of NumPy provided by PennyLane.
+
+import pennylane as qml
+from pennylane import numpy as np
+
+
+##############################################################################
+# .. important::
+#
+#     When constructing a hybrid quantum/classical computational model with PennyLane,
+#     it is important to **always import NumPy from PennyLane**, not the standard NumPy!
+#
+#     By importing the wrapped version of NumPy provided by PennyLane, you can combine
+#     the power of NumPy with PennyLane:
+#
+#     * continue to use the classical NumPy functions and arrays you know and love
+#     * combine quantum functions (evaluated on quantum hardware/simulators) and
+#       classical functions (provided by NumPy)
+#     * allow PennyLane to automatically calculate gradients of both classical and
+#       quantum functions
+
+
+##############################################################################
+# Creating a device
+# -----------------
+#
+# Before we can construct our quantum node, we need to initialize a **device**.
+#
+# .. admonition:: Definition
+#     :class: defn
+#
+#     Any computational object that can apply quantum operations, and return an measurement value
+#     is called a quantum **device**.
+#
+#     In PennyLane, a device could be a hardware device (such as the IBM QX4, via the
+#     PennyLane-PQ plugin), or a software simulator (such as Strawberry Fields, via the
+#     PennyLane-SF plugin).
+#
+# .. tip::
```

Comment:
We might as well write these things as notes, no? Makes the generated notebooks slightly less terrible

Reply:
@AroosaIjaz and I did this at one point to see what it looked like. The generated notebook still looked terrible (but slightly less so), while the doc page ended up looking messier.

The issue is with pandoc - ReST markup within a `.. note::` admonition won't get converted to markdown/html in the notebook, and looks just as ugly, but now stands out because it is in a bright blue box :laughing: 

Our thinking was that 95% of users will interact with the tutorials via the pennylane documentation/website, so rather than finding a mediocre halfway between Jupyter/ReST, focus on making the ReST version look as good as possible. (also, users downloading the notebooks will be doing so after viewing the documentation page).